### ISSUE TITLE
Cannot authenticate 

### ISSUE THREAD

**Issue by centurianii** [google/google-api-php-client#1331]
I test it and there is no way to pass the point where I make a call to `$client->authenticate($_GET['code']);`, it always returns error `cURL error 6: Could not resolve host: www.googleapis.com (see http://curl.haxx.se/libcurl/c/libcurl-errors.html). in /.../GuzzleHttp/Handler/CurlFactory.php on line 186`, where `/.../GuzzleHttp/Handler/CurlFactory.php` is a real path in my localhost to this file, see my [post](https://stackoverflow.com/questions/47022093/google-php-client-library-breaks-on-authentication-with-curl-error-6).

Any idea?

**Comment by mattwhisenhunt** [comment]
Are you trying to use a specific API or use Google OAuth for something else? For the most part I use this project with an API service, for example [Gmail].(https://developers.google.com/gmail/api/quickstart/php)

**Comment by centurianii** [comment]
Trying to make a social login. Sometimes it does work but mostly gives error as in stackoverflow [link](https://stackoverflow.com/questions/47022093/google-php-client-library-breaks-on-authentication-with-curl-error-6).
One thought I had was about composer as I use my own loader and I do not receive errors for classes not loaded from my code. I understand that some of the dependencies load static files like GuzzleHttp with `functions.php` but I do load these also. In stackoverflow link I added a full example that easily(?) can someone run **with** composer running.
At the moment I managed to **succed** in getting user logged in using **php-curl** but most of the time it needs 2 requests to get user info. What I'm talking about is a replacement of thousand lines of code-the whole library of just 40-50 lines! 
Anyway, I have to decide if this library worths the effort or I should continue with curl as my next step will be to manipulate Google drive (with curl?).
